app/src/main/java/com/example/footprnt/Profile/ProfileFragment.java

```diff
@@ -173,53 +173,37 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     /**
-     * Helper method to set up the toolbar
+     * Helper method to set up the toolbar. Toolbar functionality includes settings
+     * log out, and viewing the users saved posts/saved activities
      *
      * @param v this view
      */
     private void setUpToolbar(final View v) {
-        // Log out button
         final ImageView settings = v.findViewById(R.id.settings);
         settings.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Object privacySetting = ParseUser.getCurrentUser().get(AppConstants.privacy);
-                if (privacySetting == null) {
-                    mIsPrivate = false;
-                } else {
-                    if ((Boolean) privacySetting == true) {
-                        mIsPrivate = true;
-                    } else {
-                        mIsPrivate = false;
-                    }
-                }
                 PopupMenu popup = new PopupMenu(getActivity(), settings);
                 popup.getMenuInflater().inflate(R.menu.popup_menu, popup.getMenu());
-                if (mIsPrivate) {
-                    popup.getMenu().getItem(1).setChecked(true);
-                } else {
-                    popup.getMenu().getItem(1).setChecked(false);
-                }
                 popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
                     public boolean onMenuItemClick(MenuItem item) {
                         if (item.getItemId() == R.id.logout) {
+                            // Logout
                             // Destroy instances of DB's on logout
                             StatDatabase.getStatDatabase(getContext()).clearAllTables();
                             PostDatabase.getPostDatabase(getContext()).clearAllTables();
                             UserDatabase.getUserDatabase(getContext()).clearAllTables();
                             ParseUser.logOut();
                             Intent intent = new Intent(getActivity(), LoginActivity.class);
                             startActivity(intent);
-                        } else {
-                            if (mIsPrivate) {
-                                item.setChecked(false);
-                                ParseUser.getCurrentUser().put(AppConstants.privacy, false);
-                                ParseUser.getCurrentUser().saveInBackground();
-                            } else {
-                                item.setChecked(true);
-                                ParseUser.getCurrentUser().put(AppConstants.privacy, true);
-                                ParseUser.getCurrentUser().saveInBackground();
-                            }
+                        } else if (item.getItemId() == R.id.savedPosts) {
```

Comment:
This code can be simplified to

Intent intent = null;
if (item.getItemId() == R.id.logout) {
  intent = ...
} else if (item.getItemId() == R.id.savedPosts) {
  intent = ...
} else if (item.getItemId() == R.id.savedActivities) {
  intent = ...
}

if (intent != null) {
  startActivity(intent);
}